dsl/method.go

```diff
@@ -36,3 +36,25 @@ func Method(name string, fn func()) {
 	ep := &expr.MethodExpr{Name: name, Service: s, DSLFunc: fn}
 	s.Methods = append(s.Methods, ep)
 }
+
+// Deprecated must appear in a Method expression.
+//
+// Method takes no arguments. If called, it assumes that
```

Comment:
Maybe something like this is slightly clearer and follows a bit more the convention (explanation then when it can appear then arguments).
```go
// Deprecated marks HTTP routes as deprecated in the generated OpenAPI specifications.
//
// Deprecated must appear in a Method HTTP expression.
// 
// Deprecated takes no argument.
```